src/app/login-form/login-form-register.component.ts

```diff
@@ -0,0 +1,58 @@
+import {Component, Inject} from '@angular/core';
+import {MAT_DIALOG_DATA, MatDialogRef} from '@angular/material';
+import {User, UserService} from '../api/user.service';
+
+import {FormControl, Validators} from '@angular/forms';
+import { MatSnackBar } from '@angular/material';
+
+@Component({
+  selector: `app-login-form-register.component`,
+  templateUrl: 'login-form-register.component.html',
+  styleUrls: ['./login-form.component.scss'],
+})
+export class LoginFormRegisterComponent {
+  constructor(public dialogRef: MatDialogRef<LoginFormRegisterComponent>,
+              private userService: UserService,
+              public snackBar: MatSnackBar,
+              @Inject(MAT_DIALOG_DATA) public data: any) {
+  }
+
+  usernameRegisterText: string;
+  passwordRegisterText: string;
+  emailText: string;
+  users: User = {
+    id: 0,
+    username: '',
+  };
+  hideInvalidText = true;
+
+  usernameFormControl = new FormControl('', [
+    Validators.required,
+  ]);
+
+  passwordFormControl = new FormControl('', [
+    Validators.required,
+  ]);
+
+  registerValidate() {
+    // Validate user registration by calling signup from UserService
+    this.userService.signup(this.usernameRegisterText, this.passwordRegisterText, this.emailText).then( (user) => {
+      this.users = user;
+      this.snackBar.open('Welcome to MemePlace!', 'close', {
+        duration: 3000
+      });
+      this.dialogRef.close();
+    }).catch((err) => {
+      this.hideInvalidText = false;
+      this.snackBar.open('Registration Failed', 'close', {
```

Comment:
You can change the text for this to the error message with `err.toString()`. It should provide more details on it.

Reply:
There's the wrong error reference passed https://github.com/MemePlace/Frontend/pull/35.

Once we merge that and you update from master, it should work.